OpenRA.Game/Scripting/ScriptObjectWrapper.cs

```diff
@@ -30,10 +30,11 @@ public ScriptObjectWrapper(ScriptContext context)
 
 		protected void Bind(IEnumerable<object> clrObjects)
 		{
+			bool isAIScriptBinding = Context is AIScriptContext;
```

Comment:
A much simpler solution is to split those mixed cases into two classes, one with ai safe things, and one without.